openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -452,6 +463,15 @@ def post(self, request, lib_key_str):
         serializer = LibraryXBlockCreationSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
         parent_block_usage_str = serializer.validated_data.pop("parent_block", None)
+        lib = api.get_library(library_key)
+        if lib.type != COMPLEX:
+            block_type = serializer.validated_data.get('block_type')
+            if serializer.validated_data.get('block_type') != lib.type:
+                raise ValidationError(
+                    _('Block type "{block_type}" is not compatible with library type "{library_type}".').format(
+                        block_type=block_type, library_type=lib.type,
+                    )
+                )
```

Comment:
Would it make sense to do this validation at the `api.py` layer, such that it would apply to any view that might add a block to a library?

Reply:
Yes. I've now moved this into the `api.py` layer.